app/src/main/java/sdp/moneyrun/game/GameBuilder.java

```diff
@@ -0,0 +1,145 @@
+package sdp.moneyrun.game;
+
+import android.location.Location;
+
+import java.util.List;
+
+import sdp.moneyrun.map.Coin;
+import sdp.moneyrun.map.Riddle;
+import sdp.moneyrun.player.Player;
+
+public class GameBuilder {
+    private String name;
+    private Player host;
+    private List<Player> players;
+    private int maxPlayerCount = 0;
+    private List<Coin> coins;
+    private List<Riddle> riddles;
+    private Location startLocation;
+    private boolean isVisible = true;
+
+    /**
+     * Empty constructor
+     */
+    public GameBuilder(){}
+
+    /**
+     * @param name the game name
+     */
+    public void setName(String name){
+        if(name == null){
+            throw new IllegalArgumentException("name should not be null.");
+        }
+
+        this.name = name;
+    }
+
+    /**
+     * @param host the game host
+     */
+    public void setHost(Player host){
```

Comment:
I think it would be good to return this instance so that the setters can be chained on the builder. Otherwise it would be tedious to write each builder.set(...); builder.set(...);  instead of just having builder.set(...).set(...)

Reply:
Yes I agree, this would actually make the code more cleaner

Reply:
done